be/src/olap/schema_change.cpp

```diff
@@ -647,12 +661,29 @@ Status VSchemaChangeWithSorting::_external_sorting(vector<RowsetSharedPtr>& src_
     Merger::Statistics stats;
     RETURN_IF_ERROR(Merger::vmerge_rowsets(new_tablet, ReaderType::READER_ALTER_TABLE,
                                            *new_tablet_schema, rs_readers, rowset_writer, &stats));
+    LOG_INFO("lightman _external_sorting").tag("merged_rows", stats.merged_rows)
+            .tag("src_rowsets len", src_rowsets.size());
     _add_merged_rows(stats.merged_rows);
     _add_filtered_rows(stats.filtered_rows);
     return Status::OK();
 }
 
-Status SchemaChangeHandler::process_alter_tablet_v2(const TAlterTabletReqV2& request) {
+Status VLocalSchemaChangeWithSorting::_inner_process(RowsetReaderSharedPtr rowset_reader, RowsetWriter* rowset_writer,
+                          BaseTabletSPtr new_tablet, TabletSchemaSPtr base_tablet_schema,
+                          TabletSchemaSPtr new_tablet_schema) {
+    Defer defer {[&]() {
+        // remove the intermediate rowsets generated by internal sorting
```

Comment:
1. The request is used in construct of SchemaChangeJob, and checked here?
2. _base_tablet is set in construct of schemachangejob, and checked here?

Reply:
_base_tablet and _new_tablet is the schemachangejob member, it will be set in ctor, but it maybe nullptr, and the ctor cannot return it. So we handle the nullptr when we call 'process_alter_tablet'